q2_metadata/normalization/_norm_rules.py

```diff
@@ -0,0 +1,322 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2020, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import yaml
+from glob import glob
+import pandas as pd
+from os.path import basename, isdir, splitext
+
+from q2_metadata.normalization._norm_messages import warningsCollection, errorsCollection
+
+
+class Rules(object):
+    """Collect the rules of one current variable rules.
+
+    Parameters
+    ----------
+    variable_rules_fp : str
+        Path to one variable's rules yaml file.
+
+    Attributes
+    ----------
+    variable : str
+        Name of the variable which rules are parsed and
+        that is the yaml file name without the .yml extension.
+
+    remap : dict
+        Apply the rules of remapping variables values, by replacing
+        all instances of a given value by another one value.
+                    'remap:
+                      str: str'
+    validation : dict
+        Apply the rules of validating variables values, by replacing by the blank
+        (or the missing) value every instance which in one or many other columns
+        have a null or missing value.
+                    'validation:
+                      force_to_blank_if:
+                        is null:
+                        - variable'
+    normalization : dict
+        Apply the rules of normalizing the variables values, by replacing
+        the values outside the range by a given replacement value.
+                    'normalization:
+                        gated_value:
+                            - Out of bounds
+                        maximum:
+                            - int
+                            - float
+                        minimum:
+                            - int
+                            - float'
+    missing : str
+        Apply the rules of checking the allowed missing values, by replacing
+        the not-expected values and numpy's nan (empty) by the passed
+        missing value term.
+                    'missing:
+                        - not applicable
+                        - not collected
+                        - not provided
+                        - restricted access'
+    format : str
+        Apply the rules of checking that the passed column in the correct format.
+        [THIS COMMAND WOULD EITHER INFER THE DTYPE (as of now) OR USE THE
+        RESULT OF AN INITIAL DTYPE GETTER FUNCTION (see _dtypes.py)]
```

Comment:
not sure i understand, is `format` a `dtype`?

Reply:
the format rule is something you defined with Rebecca a while back - to my understanding, it indeed is some sort of dtype, but from a user-end, it could be "str", "float", "int".. I have ever created a "yyyy-mm-dd" for a date formatting.  That's open to discussion but remember that there's already tests for this (although not problem to edit too).

also here - this was not an attribute - it is now indented as a docstring intro text.